tools/serverpod_cli/lib/src/generator/dart/library_generators/library_generator.dart

```diff
@@ -1614,3 +1616,33 @@ extension on DatabaseDefinition {
     ]).code;
   }
 }
+
+/// Sorts child classes before their parents, such that serialization order is stable.
+int _byChildClassesBeforeParents(
+  SerializableModelDefinition a,
+  SerializableModelDefinition b,
+) {
+  if (a is! ModelClassDefinition || b is! ModelClassDefinition) {
+    return 0;
+  }
+
+  if (a.extendsClass != null && b.extendsClass == null) {
+    return -1;
+  }
+
+  if (a.extendsClass == null && b.extendsClass != null) {
+    return 1;
+  }
+
+  if (a.extendsClass is ResolvedInheritanceDefinition &&
```

Comment:
could use new if `if (x case T)` syntax to avoid later cast

Reply:
Right, that should be available now :)